crates/handlers/src/graphql/mutations/user.rs

```diff
@@ -199,6 +200,66 @@ impl AllowUserCrossSigningResetPayload {
     }
 }
 
+/// The input for the `setPassword` mutation.
+#[derive(InputObject)]
+struct SetPasswordInput {
+    /// The ID of the user to set the password for.
+    /// If you are not a server administrator then this must be your own user
+    /// ID.
+    user_id: ID,
+
+    /// The current password of the user.
+    /// Required if you are not a server administrator.
+    current_password: Option<String>,
+
+    /// The new password for the user.
+    new_password: String,
+}
+
+/// The return type for the `setPassword` mutation.
+#[derive(Description)]
+struct SetPasswordPayload {
+    status: SetPasswordStatus,
```

Comment:
In other mutations, we return the object which changed, so that the app can directly things which changed in the mutation without doing another round trip.
Example of such a thing: https://github.com/matrix-org/matrix-authentication-service/blob/a63ba4cb8f2e9d756ecec0323e46877ee474f239/crates/graphql/src/mutations/user.rs#L153-L172

buuut since we're not yet exposing whether the user has a password or not, or when it was last updated, there is no updated data to give for now, so I'm fine with adding that later

Reply:
Yes, also sending the password hash back to the user doesn't seem like a desirable thing anyway ;-)